Tests/Core/AWSClientRuntimeTests/Auth/AuthSchemeTests/SigV4AuthSchemeTests.swift

```diff
@@ -0,0 +1,285 @@
+            .build()
+        let updatedProperties = try sigV4AuthScheme.customizeSigningProperties(signingProperties: Attributes(), context: context)
+        XCTAssertEqual(AWSSignedBodyHeader.contentSha256, updatedProperties.get(key: AttributeKeys.signedBodyHeader))
+    }
+
+    func testUseSignedBodyHeaderWhenServiceIsGlacierAndUseUnsignedBodyIsFalse() throws {
+        let context = contextBuilder
+            .with(key: AttributeKeys.bidirectionalStreaming, value: false)
+            .withServiceName(value: "Glacier")
+            .withFlowType(value: .NORMAL)
+            .withOperation(value: "filler")
+            .withUnsignedPayloadTrait(value: false)
+            .build()
+        let updatedProperties = try sigV4AuthScheme.customizeSigningProperties(signingProperties: Attributes(), context: context)
+        XCTAssertEqual(AWSSignedBodyHeader.contentSha256, updatedProperties.get(key: AttributeKeys.signedBodyHeader))
+    }
+
+    func testUseSignedBodyHeaderWhenServiceIsS3ControlAndUseUnsignedBodyIsFalse() throws {
+        let context = contextBuilder
+            .with(key: AttributeKeys.bidirectionalStreaming, value: false)
+            .withServiceName(value: "S3 Control")
+            .withFlowType(value: .NORMAL)
+            .withOperation(value: "filler")
+            .withUnsignedPayloadTrait(value: false)
+            .build()
+        let updatedProperties = try sigV4AuthScheme.customizeSigningProperties(signingProperties: Attributes(), context: context)
+        XCTAssertEqual(AWSSignedBodyHeader.contentSha256, updatedProperties.get(key: AttributeKeys.signedBodyHeader))
+    }
+
+    func testDontUseSignedBodyHeaderWhenUseUnsignedBodyIsTrue() throws {
+        let context = contextBuilder
+            .with(key: AttributeKeys.bidirectionalStreaming, value: false)
+            .withServiceName(value: "S3")
+            .withFlowType(value: .NORMAL)
+            .withOperation(value: "filler")
+            .withUnsignedPayloadTrait(value: true)
+            .build()
+        let updatedProperties = try sigV4AuthScheme.customizeSigningProperties(signingProperties: Attributes(), context: context)
+        XCTAssertEqual(AWSSignedBodyHeader.none, updatedProperties.get(key: AttributeKeys.signedBodyHeader))
+    }
+
+    // S3-specific customizations
+
+    // AttributeKeys.useDoubleURIEncode flag
+
+    func testUseDoubleURIEncodeIsFalseWhenServiceIsS3() throws {
+        let context = contextBuilder
+            .with(key: AttributeKeys.bidirectionalStreaming, value: false)
+            .withServiceName(value: "S3")
+            .withFlowType(value: .NORMAL)
+            .withOperation(value: "filler")
+            .withUnsignedPayloadTrait(value: false)
+            .build()
+        let updatedProperties = try sigV4AuthScheme.customizeSigningProperties(signingProperties: Attributes(), context: context)
+        XCTAssertEqual(false, updatedProperties.get(key: AttributeKeys.useDoubleURIEncode))
```

Comment:
When comparing to `true` or `false`, use `XCTAssertTrue()` or `XCTAssertFalse()`

Reply:
Changed as suggested.